main.js

```diff
@@ -191,6 +196,9 @@ app.on('ready', function() {
     // Initialise window mgr
     Windows.init();
 
+    // check for update
+    require('./modules/updateChecker').run();
```

Comment:
Why are you checking for the update in the backend and not in the update check window?

I would argue that this just adds unnecessary complexity to the backend and extra pic connections, which might lead to memory leaks (i try to keep these kind of connections as less as possible.)